### ISSUE TITLE
Fix failures in ETW logging on 4GB aware 32 bit processes (second try)

### ISSUE THREAD

**Issue by vancem** [dotnet/coreclr#12015]
PR #11941 was a first attempt at a fix, but it was not correct.  

@sharwell notices this through review (good job), but the fix was already merged.   This issue tracks following up on the correct fix and testing.

**Comment by vancem** [comment]
The good news is the fix is trivial.  

            public unsafe IntPtr DataPointer { get { return (IntPtr)m_Ptr; } set { m_Ptr = unchecked((ulong)(void*)value); } }

Need to do testing however.  

@brianrob

**Comment by sharwell** [comment]
:memo: I think the getter needs to be updated as well, as can be shown by the following:

```
> ulong ptr = 0xdeadf00d;
> (IntPtr)ptr
Arithmetic operation resulted in an overflow.
  + System.IntPtr.op_Explicit(long)
```

For that, I think you need this:

```csharp
return (IntPtr)(void*)m_Ptr;
```

**Comment by vancem** [comment]
This fix has been merged in PR #12022, we can close the issue.
